### ISSUE TITLE
error

### ISSUE THREAD

**Issue by GuangtaoLyu** [HemantKumar01/AIPersonalizedLearningPlatform#1]
Unexpected Application Error!
Cannot read properties of undefined (reading 'map')

hello, thank you for your great work.
i try to run your code.
but when i create a new roadmap, this error occured.
how to solve it?
thank you.

**Comment by GuangtaoLyu** [comment]
ERROR
Cannot read properties of undefined (reading 'map')
TypeError: Cannot read properties of undefined (reading 'map')
    at TopicBar (http://localhost:3000/static/js/bundle.js:2066:31)
    at renderWithHooks (http://localhost:3000/static/js/bundle.js:36920:22)
    at mountIndeterminateComponent (http://localhost:3000/static/js/bundle.js:40891:17)
    at beginWork (http://localhost:3000/static/js/bundle.js:42194:20)
    at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:27176:18)
    at Object.invokeGuardedCallbackDev (http://localhost:3000/static/js/bundle.js:27220:20)
    at invokeGuardedCallback (http://localhost:3000/static/js/bundle.js:27277:35)
    at beginWork$1 (http://localhost:3000/static/js/bundle.js:47175:11)
    at performUnitOfWork (http://localhost:3000/static/js/bundle.js:46423:16)
    at workLoopSync (http://localhost:3000/static/js/bundle.js:46346:9)

**Comment by HemantKumar01** [comment]
@GuangtaoLyu 
I tried to reproduce the error, but apparently everything is working fine for me.
make sure to add `GEMINI_API_KEY` by creating a .env file in backend folder. 
Provide me more details on when exactly the error occurred (were you able to start the frontend and backend?).
I will try to fix the error as soon as possible.

**Comment by GuangtaoLyu** [comment]
> @GuangtaoLyu I tried to reproduce the error, but apparently everything is working fine for me. make sure to add `GEMINI_API_KEY` by creating a .env file in backend folder. Provide me more details on when exactly the error occurred (were you able to start the frontend and backend?). I will try to fix the error as soon as possible.

![image](https://github.com/user-attachments/assets/41fe023f-5ac8-478e-b9c6-c5829cc055dc)
I can start the frontend and backend. 
but when I click the start learning button. the error happend.

**Comment by GuangtaoLyu** [comment]
> @GuangtaoLyu I tried to reproduce the error, but apparently everything is working fine for me. make sure to add `GEMINI_API_KEY` by creating a .env file in backend folder. Provide me more details on when exactly the error occurred (were you able to start the frontend and backend?). I will try to fix the error as soon as possible.

![image](https://github.com/user-attachments/assets/910d1404-ead2-4d18-9217-cacda920af50)

API works well.

**Comment by HemantKumar01** [comment]
@GuangtaoLyu thanks for bringing the issue to my attention
Fixed the error in the latest commit. 
install the pip packages again using the new requirements.txt, and run the frontend and backend again and it should work now.
(or clone the updated repo and do a fresh setup)

The error was only happening after `npm run build` because of proxy not present there.
The proxy server, as defined in `package.json` works only for development server. To make it work for now I hardcoded `localhost:5000` for api requests. 
If you are hosting the backend somewhere else, make sure to replace `localhost:5000` with your backend url everywhere in frontend.
